presto-docs/src/main/sphinx/security/salesforce.rst

```diff
@@ -0,0 +1,78 @@
+username and password (or password and :ref:`security token` concatenation)
+are validated by having the Presto coordinator perform a login to Salesforce.
+
+This allows Presto admins to enable users to authenticate to Presto via their Salesforce
+basic credentials. This is *not* a Salesforce connector, and does not allow users to query
+Salesforce data. It's simply a means by which users can authenticate to Presto, similar
+to :doc:`ldap` or :doc:`password-file`.
+
+Salesforce Authenticator Configuration
+--------------------------------------
+
+Enable Salesforce authentication by creating an
+``etc/password-authenticator.properties`` file on the coordinator:
+
+.. code-block:: none
+
+    password-authenticator.name=salesforce
+    salesforce.allowed-organizations=allowed salesforce org id's, or the word all
+
+The following configuration properties are available:
+
+====================================   ==============================================
+Property                               Description
+====================================   ==============================================
+``salesforce.allowed-organizations``   Comma separated list of 18 character Salesforce.com
+                                       Organization ID's for a second simple layer of security.
+                                       This option can be explicitly ignored using ``all``
+                                       here, which will bypass any check of the authenticated
+                                       user's Salesforce Organization Id.
+
+``salesforce.cache-size``              Maximum size of the cache that holds authenticated users.
+                                       Defaults to ``4096``.
+
+``salesforce.cache-expire-duration``   Expire duration for an entry in cache since last write.
+                                       Defaults to ``2m``.
+====================================   ==============================================
+
+Salesforce Concepts
+-------------------
+
+.. _security token:
+
+Security Token
+^^^^^^^^^^^^^^
+
+Credentials are a user's Salesforce username and password if Presto is connecting from a whitelisted
+IP, or username and password / security token concatenation otherwise.  For example, if the coordinator
+is not whitelisted, and your password is ``password`` and security token is ``token``, use ``passwordtoken``
+as your Presto password.
+
+To learn more about the Salesforce security token, see the `reset security token documentation
+<https://help.salesforce.com/articleView?id=user_security_token.htm&type=5>`_.
+
+To learn more about whitelisting an IP range, see the `set trusted IP ranges documentation
+<https://help.salesforce.com/articleView?id=security_networkaccess.htm&type=5>`_.  Use this to
+procedure to trust the public IP for the Presto coordinator.
+
+Salesforce Organization Id's
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There is a secondary layer of security beyond simply providing credentials, which is that
```

Comment:
You can configure a one or more 18 character Salesforce.com organization IDs. They can be used as a secondray layer of security.

Reply:
Will update.